stable/artifactory-ha/templates/artifactory-binarystore.yaml

```diff
@@ -42,6 +43,38 @@ data:
             <property name="header.remote.block" value="true"/>
         </provider>
     </config>
+    {{- end }}
+{{- end }}
+
+{{- if eq .Values.artifactory.persistence.type "file-system" }}
```

Comment:
They are essentially 2 completely different approaches. One is for the sharding-cluster (which additionally uses dynamic remote provider) and the other is just "sharding" which only uses state-aware. I figured it would be too many if blocks inside if I tried to piecemeal them together..  but maybe what I'm hearing is something more like this:

```
{{- if eq .Values.artifactory.persistence.type "file-system" }}
    <!-- File system replication -->
    <config version="2">
         {{- if .Values.artifactory.persistence.fileSystem.existingSharedClaim.enabled }}
            existing shared configs here
         {{ else }}
            normal dynamic configs here
         {{ end }}
    </config>
{{- end }}
```

Reply:
yeah that's what I meant. Thanks @zendril

Reply:
Done.